src/UI/Art/SpriteArtist/SpriteLibrary.java

```diff
@@ -318,20 +320,24 @@ public static UnitSpriteSet getMapUnitSpriteSet(String type, Faction faction, Co
   private static void createMapUnitSpriteSet(UnitSpriteSetKey key)
   {
     Faction faction = key.factionKey;
-    String filestr = getMapUnitSpriteFilename(key.unitTypeKey, faction.name);
-    if (!new File(filestr).canRead())
-      filestr = getMapUnitSpriteFilename(key.unitTypeKey, faction.basis);
-    UnitSpriteSet spriteSet = new UnitSpriteSet(SpriteLibrary.loadSpriteSheetFile(filestr), baseSpriteSize, baseSpriteSize,
-        UIUtils.getMapUnitColors(key.colorKey));
+
+    Function<AnimState,String> fileFinder = (state) -> getMapUnitSpriteFilename(key.unitTypeKey, faction.name, state);
+    if (!new File(getMapUnitSpriteFilename(key.unitTypeKey, faction.name)).canRead())
+      fileFinder = (state) -> getMapUnitSpriteFilename(key.unitTypeKey, faction.basis, state);
+
+    UnitSpriteSet spriteSet = new UnitSpriteSet( fileFinder, UIUtils.getMapUnitColors(key.colorKey) );
     mapUnitSpriteSetMap.put(key, spriteSet);
   }
 
   private static String getMapUnitSpriteFilename(String unitType, String faction)
-  {
-    StringBuffer spriteFile = new StringBuffer();
-    spriteFile.append("res/unit/faction/").append(faction).append("/");
-    spriteFile.append(UnitModel.standardizeID(unitType)).append("_map.png");
-    return spriteFile.toString();
+  { return getMapUnitSpriteFilename(unitType, faction, AnimState.IDLE); }
+  private static String getMapUnitSpriteFilename(String unitType, String faction, AnimState state)
+  {
+    final String format = "res/unit/faction/%s/%s_map%s.png";
```

Comment:
Ah, so you have both lambdas *and* templated strings.I still feel like this could be simplified.

Reply:
/shrug